app.rb

```diff
@@ -0,0 +1,97 @@
+require './book'
+require './teacher'
+require './rental'
+require './student'
+class App
+  def initialize(opt)
+    @persons = []
+    @books = []
+    @rentals = []
+    @option = opt
+  end
+
+  def list_all_books
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    @option.list_options
+  end
+
+  def list__all_people
+    @persons.each { |person| puts "ID: #{person.id}, Name: #{person.name}, Age: #{person.age}" }
+    @option.list_options
+  end
+
+  def create_person
+    # 1 for student and 2 for teacher
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    profession = gets.to_i
+    if [1, 2].include? profession
+      print 'Age: '
+      age = gets.to_i
+      print 'Name: '
+      name = gets.chomp
+      if profession == 1
+        create_student(name, age)
+      else
+        create_teacher(name, age)
+      end
+      puts 'Person created successfully'
+      @option.list_options
+    else
```

Comment:
- [ ] Great work so far, however I think there is a bug that need your attention currently when we create a person whether it is teacher or student, the input we provide for parent_permission attribute for user and specialisation attribute for teacher comes out in the value of age when we list all people, I have recorder a short video demo below please see and try to fix this :point_down: 

https://user-images.githubusercontent.com/97521557/233792445-8d188d72-1fc5-4e0f-a5a4-e42310718273.mp4

Reply:
Hello @PinkMoon25 thank you for your remark I will correct it now